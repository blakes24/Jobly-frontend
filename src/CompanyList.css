.CompanyList {
  max-width: 800px;
  margin: 0 auto;
}

.CompanyList form {
  display: flex;
  justify-content: stretch;
  margin: 10px;
}
