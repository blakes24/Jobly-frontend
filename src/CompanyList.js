import { useState, useEffect } from 'react';
import { Form, Button } from 'react-bootstrap';
import JoblyApi from './api';
import CompanyCard from './CompanyCard';

const CompanyList = () => {
  const [ list, setList ] = useState([]);
  const [ isLoading, setIsLoading ] = useState(true);
  const [ formData, setFormData ] = useState({ search: '' });

  // get companies when page loads and add to state
  useEffect(() => {
    async function getList() {
      const list = await JoblyApi.getCompanies();
      setList(list);
      setIsLoading(false);
    }
    getList();
  }, []);

  async function filter(params) {
    const list = await JoblyApi.getCompanies(params);
    setList(list);
  }

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      [name] : value
    }));
  };

  // Filter list based on search term
  const handleSubmit = (evt) => {
    evt.preventDefault();
    // if search is empty reset and get all companies
    if (formData.search.trim().length < 1) {
      setFormData((formData) => ({
        ...formData,
        search : ''
      }));
      filter();
    } else {
      filter(formData.search);
    }
  };

  return (
    <div>
      <Form onSubmit={handleSubmit} inline>
        <Form.Control
          className="my-2"
          id="search"
          name="search"
          placeholder="Enter search term"
          type="text"
          value={formData.search}
          onChange={handleChange}
        />
        <Button type="submit" className="my-2">
          Submit
        </Button>
      </Form>
      {isLoading ? (
        <h2>Loading...</h2>
      ) : list.length ? (
        list.map((company) => (
          <CompanyCard
            name={company.name}
            description={company.description}
            logoUrl={company.logoUrl}
            handle={company.handle}
            key={company.handle}
          />
        ))
      ) : (
        <p>No results found</p>
      )}
    </div>
  );
};

export default CompanyList;
