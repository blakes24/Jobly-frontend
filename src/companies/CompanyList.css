.CompanyList {
  max-width: 800px;
  margin: 0 auto;
}

.CompanyList form {
  display: flex;
  justify-content: stretch;
  margin: 10px;
}

.CompanyList > a {
  margin: 10px;
  font-size: x-small;
}

.CompanyList .spinner-border {
  display: block;
  margin: 100px auto;
  color: #015dc0;
}
