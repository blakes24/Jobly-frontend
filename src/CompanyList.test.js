import { render } from '@testing-library/react';
import CompanyList from './CompanyList';

test('renders without crashing', () => {
  render(<CompanyList />);
});

it('matches snapshot', function() {
  const { asFragment } = render(<CompanyList />);
  expect(asFragment()).toMatchSnapshot();
});
