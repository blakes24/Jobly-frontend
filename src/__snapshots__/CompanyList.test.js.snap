// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyList"
  >
    <form
      class=""
    >
      <input
        class="my-2 form-control"
        id="search"
        name="search"
        placeholder="Enter search term"
        type="text"
        value=""
      />
      <button
        class="my-2 btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <h2>
      Loading...
    </h2>
  </div>
</DocumentFragment>
`;
